Documentation/planning/multi-sdk-band-support.md

```diff
@@ -0,0 +1,253 @@
+</ItemGroup>
+
+<PropertyGroup>
+    <UseBootstrapArcade Condition="'$(UseBootstrapArcade)' == '' and '@(ActiveArcadeDependency)' == ''">true</UseBootstrapArcade>
+</PropertyGroup>
+
+<ItemGroup>
+    <UseSourceBuiltSdkOverride Condition="'$(UseBootstrapArcade)' != 'true'" Include="@(ArcadeSdkOverride)" />
+    <UseSourceBuiltSdkOverride Condition="'$(UseBootstrapArcade)' == 'true'" Include="@(ArcadeBootstrapSdkOverride)" />
+    <EnvironmentVariables Include="_InitializeToolset=$(SourceBuiltSdksDir)Microsoft.DotNet.Arcade.Sdk/tools/Build.proj" Condition="'$(UseBootstrapArcade)' != 'true'" />
+</ItemGroup>
+```
+
+***Note: Today it is necessary to explicitly set UseBootstrapArcade for those projects where arcade is not available (arcade, source-build-reference-packages). It may be possible to remove this requirement, simply relying on the repo dependency graph to give the correct answer.***
+
+### Input asset locations
+
+The input assets required for the .NET build come in two primary forms:
+- NuGet assets
+- Non-NuGet assets (typically `.zip` or `.tar.gz` archives)
+
+Source build handles NuGet assets using a set of local NuGet feeds (which are just directories of NuGet packages). It handles non-Nuget assets by using one of these local NuGet feeds as a shared storage location where repos producing archives (e.g. runtime, sdk, aspnetcore) place their outputs (`blob-feed` dir). Downstream repos that need these archives copy the files from that location. They locate specific file names based on set naming patterns combined with package input versions. For instance, the installer repo might find the runtime archive for linux-x64 by looking for `dotnet-runtime-<version of runtime package>-linux-x64.tar.gz`. When building a subsetted VMR, just as the shared location may not contain all the input packages, it may not contain all of the input archives. For NuGet packages, this is handled reasonably transparently. Given a set of potential input feeds, NuGet simply checks each one. So if arcade wasn't built and version 1.2.3 isn't available in the shared `blob-feed` dir, NuGet will look in the previously source-built artifacts feed.
+
+Because CSBs are separated from PSBs when building a subsetted VMR, source-build needs to be altered to add new CSB location as a NuGet input to each component build. In addition, builds of subsetted VMRs need a way to locate non-NuGet artifacts from the CSBs, without mixing them into the shared `blob-feed` where component builds currently look. Repositories should be altered to accept an additional location where they may find input artifacts. This has precedent, as it is the same logic used in installer, aspnetcore, windowsdesktop, and others in Microsoft's non-source-build scenarios. These repos will check a public official source for binaries first, then a public build artifacts location, and finally an internal build artifacts location, if the appropriate credentials are available.
```

Comment:
Do CSBs and PSBs contain only NuGets (intermediate packages) or also non-NuGets?

Reply:
CSBs contain non-NuGets for sure (e.g. runtime tar.gz file). PSBs could contain the non-NuGet assets, but they wouldn't be used. This is the case today. You really only need the non-NuGet (or any assets, for that matter) that would be used in the build you want to do.

For clarity and consistency, we should say that PSB and CSB artifacts should contain all output assets.